src/openrct2/entity/Guest.cpp

```diff
@@ -3644,6 +3644,10 @@ void Guest::UpdateRideAdvanceThroughEntrance()
         MoveTo({ loc.value(), actionZ });
         return;
     }
+    else if (RideSubState == PeepRideSubState::InEntrance)
+    {
+        RideSubState = PeepRideSubState::LeaveEntrance;
```

Comment:
This is unsafe should perform `FreeVehicleCheck` then immediately return from this function.